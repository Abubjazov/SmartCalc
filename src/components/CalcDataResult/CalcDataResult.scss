@import '../../variables';

.calc-data-result {
	display: grid;
	grid-template-columns: 300px;
	grid-template-rows: 50px auto 30px auto;
	grid-row: auto;
	row-gap: 10px;
	justify-items: center;

	h2 {
		color: $black;
	}

	button {
		margin-top: 30px;
		padding: 0 13px;
		border: none;
		border-radius: 20px;
		min-width: 100px;
		height: 40px;
		text-align: center;
		font-size: 14px;
		line-height: 14px;
		background-color: $main-color;
		color: $white;
		transition: all 0.1s linear;
		outline: none;
		cursor: pointer;

		&:hover,
		&:focus {
			background-color: $main-color-hover;
		}
	}

	ul {
		margin-top: 13px;
		justify-self: center;
		font-size: 20px;
		font-weight: 300;

		li {
			width: 200px;
			text-align: center;
			border-bottom: 1px solid $deep-gray;
		}

		.green {
			color: $green;
		}
	}

	&-footer {
		width: 90%;
		margin-top: 31px;
		display: flex;
		justify-content: space-between;
	}
}
