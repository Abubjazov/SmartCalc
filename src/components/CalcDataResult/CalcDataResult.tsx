import { nanoid } from 'nanoid'
import { useState } from 'react'
import { NavLink } from 'react-router-dom'

import './CalcDataResult.scss'

export const CalcDataResult = (): JSX.Element => {
	const [items] = useState<number[]>([10, 11, 9])

	return (
		<div className='calc-data-result'>
			<h1>Результаты расчёта</h1>

			<ul>
				{items.map(item => (
					<li key={nanoid()} className={item > 10 ? 'green' : ''}>
						{item}
					</li>
				))}
			</ul>

			<h2>Сумма: 39</h2>

			<div className='calc-data-result-footer'>
				<NavLink to='/smartcalc/calc'>Назад</NavLink>
				<NavLink to='/smartcalc/calc'>Новый расчёт</NavLink>
			</div>
		</div>
	)
}
