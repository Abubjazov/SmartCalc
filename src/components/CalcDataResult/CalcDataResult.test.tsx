import { render } from '@testing-library/react'
import { CalcDataResult } from './CalcDataResult'

describe('Component: Calculator', () => {
	test('should render Calculator', () => {
		const { asFragment } = render(<CalcDataResult />)

		expect(asFragment()).toMatchSnapshot()
	})
})
