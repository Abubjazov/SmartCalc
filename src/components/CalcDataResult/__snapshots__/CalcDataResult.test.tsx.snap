// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CalcDataInput should render CalcDataInput by default 1`] = `
<DocumentFragment>
  <div
    class="calc-data-result"
  >
    <h1>
      Результаты расчёта
    </h1>
    <ul>
      <li
        class=""
      >
        1
      </li>
      <li
        class=""
      >
        2
      </li>
      <li
        class=""
      >
        3
      </li>
      <li
        class="green"
      >
        44
      </li>
    </ul>
    <h2>
      Сумма: 51
    </h2>
    <div
      class="calc-data-result-footer"
    >
      <button>
        Ввод данных
      </button>
      <button>
        Новый расчёт
      </button>
    </div>
  </div>
</DocumentFragment>
`;
